geopandas/vectorized.pyx

```diff
@@ -921,14 +921,6 @@ class GeometryArray(object):
         self.data = geoms
         self.parent = None
 
-    @property
-    def x(self):
-        return get_coordinate_point(self.data, 0)
```

Comment:
I think it is OK to keep this, we just need to deal with the case when it are not all Points. I recently merged a PR of @jdmcbr to add this to GeoSeries (not yet in this branch), see https://github.com/geopandas/geopandas/pull/383/files. There we raise a ValueError if not all geoms are points.

(which of course does not mean GeometryArray should have this, as we can also put this logic in GeoSeries, and use there the `get_coordinate_point`)